downstreamadapter/sink/cloudstorage/writer.go

```diff
@@ -255,22 +255,17 @@ func (d *writer) writeDataFile(ctx context.Context, path string, task *singleTab
 			return 0, 0, inErr
 		}
 
-		defer func() {
-			closeErr := writer.Close(ctx)
-			if inErr != nil {
-				log.Error("failed to close writer", zap.Error(closeErr),
-					zap.Int("workerID", d.id),
-					zap.Any("table", task.tableInfo.TableName),
-					zap.String("keyspace", d.changeFeedID.Keyspace()),
-					zap.Stringer("changefeed", d.changeFeedID.ID()))
-				if inErr == nil {
-					inErr = closeErr
-				}
-			}
-		}()
 		if _, inErr = writer.Write(ctx, buf.Bytes()); inErr != nil {
 			return 0, 0, inErr
 		}
+		if inErr = writer.Close(ctx); inErr != nil {
+			log.Error("failed to close writer", zap.Error(inErr),
+				zap.Int("workerID", d.id),
+				zap.Any("table", task.tableInfo.TableName),
+				zap.String("keyspace", d.changeFeedID.Keyspace()),
+				zap.Stringer("changefeed", d.changeFeedID.ID()))
+			return 0, 0, inErr
```

Comment:
You must make sure this error is thrown to the maintainer and triggers changefeed restat -- this prevents the disptahcer from getting stuck due to **missing callbacks**.

Reply:
I have checked this and determined the maintainer will receive this error